src/components/molecules/AuthForm/AuthForm.tsx

```diff
@@ -19,28 +19,69 @@ interface AuthProps {
 }
 
 const AuthForm: FC<AuthProps> = ({ type }) => {
-  const { login, signup, loading } = useAuth();
+  const { login, signup, loading, error: authError, setError } = useAuth();
   const router = useRouter();
   const [formFields, setFormFields] = useState<Auth.IFormFields>({
     email: '',
     password: '',
     repeatPassword: '',
   });
 
+  const [errors, setErrors] = useState<Auth.IFormFields>({
+    email: '',
+    password: '',
+    repeatPassword: '',
+  });
+
+  const validateEmail = (email: string): boolean => {
+    const regex =
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    if (!regex.test(email)) {
+      setErrors({ ...errors, email: 'Invalid email' });
+      return false;
+    }
+    return true;
+  };
+
+  const validatePassword = (password: string): boolean => {
+    if (password.length < 8) {
```

Comment:
Always good practice to label hardcoded values like this by assigning them to a constant (put it in `src/constants/app.ts` as well)
```suggestion
    if (password.length < MIN_PASSWORD_LENGTH) {
```